setup.py

```diff
@@ -33,9 +33,11 @@
     classifiers=[
         'License :: OSI Approved :: Apache Software License',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
     ],
     python_requires='>=3.7',
```

Comment:
Are we dropping 3.7? If so, update this too please

Reply:
Ugh maybe I should just upgrade the package manager now as then I wouldn't have to do this, but scope is then getting too large, I'll update this